core/src/test/java/com/google/googlejavaformat/java/FormatterTest.java

```diff
@@ -309,4 +312,16 @@ public void stringEscapeLength() throws Exception {
     assertThat(new Formatter().formatSource("class T {{ f(\"\\\"\"); }}"))
         .isEqualTo("class T {\n  {\n    f(\"\\\"\");\n  }\n}\n");
   }
+
+  @Test
+  public void nonUnixStyleFails() throws Exception {
+    String win = "class T {\r\n}\r\n";
+    try {
+      new Formatter().formatSource(win, Collections.singleton(Range.closedOpen(0, win.length())));
+      fail("Windows-style line separator should fail!");
```

Comment:
Another nit: that exclamation mark looks a bit...  sudden! 😉

Could it be replaced with a full stop?